Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewController.swift

```diff
@@ -0,0 +1,196 @@
+//
+//  ReviewProductViewController.swift
+//  Dokan
+//
+//  Created by marwa on 16/08/2022.
+//
+
+import Domain
+import Networking
+import UIDokan
+import UIKit
+
+class ReviewProductViewController: UIViewController {
+
+    // MARK: Outlets
+
+    @IBOutlet private(set) weak var reviewProductTableView: UITableView!
+    @IBOutlet private(set) weak var mainStackView: UIStackView!
+    @IBOutlet private(set) weak var ratingDetailsView: RatingDetailsView!
+
+    // MARK: Properties
+
+    private var viewModel: ReviewProductViewModelType
+    private var navigationBarBehavior: ReviewProductNavigationBarBehavior?
+    private var reviewCount: Int = 10
+    private var loadingCell: Bool = true
+
+    // MARK: Init
+
+    init(viewModel: ReviewProductViewModelType) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        tableViewSetup()
+        bindViewModel()
+        configureNavBar()
+    }
+}
+
+// MARK: - Actions
+
+extension ReviewProductViewController {}
+
+// MARK: - Configurations
+
+extension ReviewProductViewController {
+
+    func tableViewSetup() {
+        let nib = UINib(nibName: Constants.tableViewCellName, bundle: nil)
+        reviewProductTableView.register(nib, forCellReuseIdentifier: Constants.cellReuseIdentifier)
+        reviewProductTableView.delegate = self
+        reviewProductTableView.dataSource = self
+        tableViewLayout()
+    }
+
+    func tableViewLayout() {
+        reviewProductTableView.estimatedRowHeight = reviewProductTableView.rowHeight
+        reviewProductTableView.rowHeight = UITableView.automaticDimension
+    }
+
+    func configureNavBar(rating: Double = 0.0) {
+        title = Constants.navBarTitle
+        navigationBarBehavior = ReviewProductNavigationBarBehavior(navigationItem: navigationItem)
+        navigationBarBehavior?.configure(rating, onBack: {
+            // TODO: back to previous screen
+        })
+    }
+}
+
+// MARK: - bind view model
+
+extension ReviewProductViewController {
+
+    func bindViewModel() {
+
+        viewModel.showAlertClosure = { () in
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+
+                let message = self.viewModel.alertMessage
+                self.showErrorAlert(error: message)
+            }
+        }
+
+        viewModel.showNavBarClosure = { () in
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+
+                let rating = self.viewModel.navBarRating
+                self.configureNavBar(rating: rating)
+            }
```

Comment:
Let's not used `DispatchQueue.main.async` unless it's truly needed. This will affect unit testing.